src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +111,168 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
             }
         }
 
-        private void SetContentType(ActionContext context, FileResult result)
+        private static void SetAcceptRangeHeader(ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.ContentType = result.ContentType;
+            response.Headers[HeaderNames.AcceptRanges] = AcceptRangeHeaderValue;
+        }
+
+        // Internal for testing
+        internal static PreconditionState GetPreconditionState(
```

Comment:
Should we do the checks in the following order? https://tools.ietf.org/html/rfc7232#section-6
@Tratcher